cachecontrol/controller.py

```diff
@@ -254,9 +275,9 @@ def cache_response(self, request, response, body=None,
 
         cc_req = self.parse_cache_control(request.headers)
         cc = self.parse_cache_control(response_headers)
-
-        cache_url = self.cache_url(request.url)
-        logger.debug('Updating cache with response from "%s"', cache_url)
+        cache_key = self.cache_key(request)
+        write_cache_key = self.cache_key(request, response)
```

Comment:
again as above this can probably go away if we just always create the key with authentication